src/hooks/useCopyToClipboard.ts

```diff
@@ -0,0 +1,30 @@
+import { useState } from "react";
+
+type CopiedValue = string;
+type CopyFn = (text: string) => Promise<boolean>; // Return success
+
+export const useCopyToClipboard = () => {
+  const [copiedText, setCopiedText] = useState<CopiedValue>("");
+
+  const copyToClipBoard: CopyFn = async (text) => {
+    if (!navigator?.clipboard) {
+      console.warn("Clipboard not supported");
+      return false;
+    }
+
+    try {
+      await navigator.clipboard.writeText(text);
+      setCopiedText(text);
+      return true;
+    } catch (error) {
+      console.warn("Copy failed", error);
+      setCopiedText("");
+      return false;
+    }
+  };
+
+  return {
+    copiedText,
```

Comment:
Do we really need this? We don't use it in anywhere, therefore maybe this hook needs to be a simple helper?